functions/go/render-helm-chart/third_party/sigs.k8s.io/kustomize/api/builtins/HelmChartInflationGenerator.go

```diff
@@ -211,49 +254,103 @@ func (p *HelmChartInflationGeneratorPlugin) cleanup() {
 	if p.tmpDir != "" {
 		os.RemoveAll(p.tmpDir)
 	}
+	if strings.HasPrefix(p.Repo, "oci://") && p.password != "" {
+		// log out of the registry
+		p.runHelmCommand([]string{
+			"registry",
+			"logout",
+			p.Registry,
+		})
+	}
 }
 
-func (p *HelmChartInflationGeneratorPlugin) Generate() (rm resmap.ResMap, err error) {
+func (p *HelmChartInflationGeneratorPlugin) Generate() (objects fn.KubeObjects, err error) {
 	defer p.cleanup()
 	if err = p.checkHelmVersion(); err != nil {
 		return nil, err
 	}
 	if _, exists := p.chartExistsLocally(); !exists {
-		if _, err := p.runHelmCommand(p.pullCommand()); err != nil {
+		var pullArgs []string
+		switch strings.HasPrefix(p.Repo, "oci://") {
```

Comment:
ditto. We may need more reliable way to determine whether the repo is OCI or not.

Reply:
For readability, true/false may not be a good signal to use switch/case :)

Reply:
Changed to if statements here.